src/backend/distributed/transaction/lock_graph.c

```diff
@@ -91,7 +91,7 @@ dump_global_wait_edges(PG_FUNCTION_ARGS)
 WaitGraph *
 BuildGlobalWaitGraph(void)
 {
-	List *workerNodeList = ActiveReadableNodeList();
+	List *workerNodeList = ActiveReadableWorkerNodeList();
```

Comment:
I  think according to the discussions,  we should involve coordinator to  the deadlock detection, so shouldn't this be `ActivePrimaryNodeList` ?

Reply:
done